docs/DeveloperGuide.md

```diff
@@ -1,38 +1,433 @@
+- `SemesterData`: Semester data contains all the module information pertaining to a single semester.
+- `Timetable`: Creates a timetable for the user with their selected modules and in their planning semester.
+
+### 3.3 Parser Component
+
+![Parser Class](images/parserClass.png)  
+
+
+The <code>Parser</code> component can:
+
+- return the correct command type based on user input.
+
+### 3.4 Command Component
+
+![Command Abstract Class](images/commandClass.png)
+
+The <code>Command</code> component can:
+- execute and return the command type based on the first word of the user input.
+
+Below is a table of command subclasses and their respective command type. The different command types extends from the
+Command class and are all in the command package.
+
+[//]: # (if the table is not necessary, remove it)
+| Command Word | Command Subclass                   | Intended Outcome                                                   |
+|--------------|------------------------------------|--------------------------------------------------------------------|
+| `add`        | `AddModuleCommand`                 | Adds the user input module into their timetable.                   |
+| `delete`     | `DeleteModuleCommand`              | Deletes the user input module from their timetable.                |
+| `list`       | `DisplaySelectedModuleListCommand` | Display all the module and slot selected by user                   |
+| `bye`        | `ExitCommand`                      | Exits the program.                                                 |
+| `export`     | `ExportCommand`                    | Creates a portable NUSMod link to create their timetable on NUSMod |
+| `get`        | `GetCommand`                       | Display all details about a module.                                |
+| `help`       | `HelpCommand`                      | Display all possible command words and their usage to user.        |
+| `import`     | `ImportCommand`                    | Import user's timetable from a NUSMod share timetable link.        |
+| `search`     | `SearchModuleCommand`              | Searches similar modules based on code, title, semester or level.  |
+| `semester`   | `SelectSemesterCommand`            | Selects the semester that the user want.                           |
+| `select`     | `SelectSlotCommand`                | Selects the time slot for the different lesson types.              |
+| `view`       | `ViewCommand`                      | Views the user timetable with user's selected modules.             |
+
+<!--
+##### How the feature is implemented
+
+##### Why it is implemented this way.
+
+##### Alternatives considered.
+-->
+
+#### 3.4.1 AddModuleCommand
+
+The <code>AddModuleCommand</code> class extends from the <code>Command</code> class and adds the user input module into
+their timetable.
+
+![AddModuleCommand Class](..\docs\images\addModuleCommandClass.png)
```

Comment:
The alt condition is a little misleading. Both conditions call on AddMessage() and they are linked by an activation box is a bit confusing to understand.